### DISCUSSION TITLE
I have a question for ObjectNative::Equals.

### DISCUSSION THREAD

**Discussion by Takym** [dotnet/runtime#53728] [Q&A] [answered]
Hello,

I was looking up the behavior of `object.Equals` and found the code below.

https://github.com/dotnet/runtime/blob/911640b3a891f92ff66e9c82ce65f71d203f11a2/src/coreclr/classlibnative/bcltype/objectnative.cpp#L139-L178

In the above code, why does not use `pThisMT` since line 166? Is there a special action in the function `pThisRef->GetMethodTable()`?

Thanks

**Comment by jkotas** [comment, marked as answer]
You are right that the code can use `pThisMT` instead of `pThisRef->GetMethodTable()`. You are welcomed to submit PR to fix it.

**Reply by Takym** [reply]
Thank you for replying. I submitted PR #54138.
